code/go/pkg/specerrors/filter.go

```diff
@@ -23,25 +23,33 @@ type Filter struct {
 type FilterResult struct {
 	Processed error
 	Removed   error
+
+	UnusedProcessors []Processor
```

Comment:
Added detection of unused rules, used by now only on tests, we could warn in the future about this.